test/live.test.ts

```diff
@@ -214,6 +240,81 @@ describe.skipIf(typeof EdgeRuntime === 'string' || typeof document !== 'undefine
       }
     })
 
+    test('handles CORS errors', async () => {
+      expect.assertions(3)
+      const restoreFetch = global.fetch
+
+      global.fetch = async (info, init) => {
+        const response = await restoreFetch(info, init)
+        if (!response.headers.has('access-control-allow-origin')) {
+          throw new Error('CORS preflight request failed')
+        }
+        return response
+      }
```

Comment:
Node.js' `global.fetch` is based on `undici`, which doesn't actually implement CORS, so we have to resort t some trickery to mimic the behaviour we want.